public/app/features/query/state/updateQueries.ts

```diff
@@ -26,8 +27,23 @@ export async function updateQueries(
     else if (currentDS && nextDS.importQueries) {
       nextQueries = await nextDS.importQueries(queries, currentDS);
     }
-    // Otherwise clear queries
+    // Otherwise clear queries that do not match the next datasource UID
     else {
+      if (currentDS) {
+        const reducedQueries: DataQuery[] = [];
+        const nextDsQueries = queries.reduce((reduced, currentQuery) => {
```

Comment:
Nit: I think it would be great to add a comment on what is the criteria to preserve the relevant comments 😄

Reply:
Added a comment 😊